ee/cis/macos-13/cis-policy-queries.yml

```diff
@@ -738,6 +738,38 @@ spec:
 ---
 apiVersion: v1
 kind: policy
+spec:
+  name: CIS - Ensure Security Auditing Is Enabled
+  platforms: macOS
+  platform: darwin
+  description: |
+    macOS's audit facility, auditd, receives notifications from the kernel when certain system calls, such as open, fork, and exit, are made. These notifications are captured and written to an audit log.
+  resolution: |
+    Automated method:
+    Ask your system administrator to deploy the following script which will enable security auditing:
+    /usr/bin/sudo /bin/launchctl load -w
+    /System/Library/LaunchDaemons/com.apple.auditd.plist
+  query: |
+    SELECT 1 where exists (
+        SELECT 
+            l.program, l.label, l.program_arguments,  
+            p.path, p.name , p.cmdline  
+        FROM
+            launchd AS l
+            INNER JOIN processes AS p
```

Comment:
What column is it using to match the entries of the two tables? (I'm used to seeing it with the `ON` syntax, something like `INNER JOIN A ON A.ID = B.ID`)

Reply:
You mean something like:

`ON (l.program = p.path) ` ?

Isn't it the same as doing this:
```
        where 
            (l.program = p.path) 
            AND
            (l.label = "com.apple.auditd") 
            AND 
            (l.program_arguments = p.cmdline)

```

Reply:
Not sure.

See the example here https://www.sqlitetutorial.net/sqlite-inner-join/, it uses `ON` to specify which column to use for joining. Maybe by default it uses the first column with the same name in both tables?

Reply:
This works as well:

`SELECT 1 where exists (
        SELECT 
            l.program, l.label, l.program_arguments,  
            p.path, p.name , p.cmdline  
        FROM
            launchd AS l
            INNER JOIN processes AS p
        ON (l.program = p.path) 
        where 
            (l.label = "com.apple.auditd") 
            AND 
            (l.program_arguments = p.cmdline)
    );`

Reply:
OK, let's add the ` ON (l.program = p.path)`.

If we don't add the ON then it might be joining all rows of one table with all rows of the other table, and then doing the `WHERE` constraints on them.

I just tested the query from the yaml without the `ON` and the query was denylisted due to exceeding performance metrics.

E.g. the following query:
```
SELECT 
    l.program, l.label, l.program_arguments,  
    p.path, p.name , p.cmdline  
FROM
    launchd AS l
    INNER JOIN processes AS p 
WHERE label = 'com.apple.auditd';
```
returns 627 results, which was the number of processes running on the system, so it seems it's just merging the two tables and then running the WHERE constraints.

Let me know if it makes sense.

Reply:
Added 👍